### ISSUE TITLE
False positive redefined-loop-name (PLW2901) with in-place augmented assignment

### ISSUE THREAD

**Issue by mihaic** [astral-sh/ruff#7273]
Considering the following contents of `issue.py` (no other files, no `pyproject.toml`):
```python
mutables = ([0, 1, 2], [10, 11, 12])
for mutable in mutables:
    mutable += [100]
```

PLW2901 is detected:
```
ruff check --select PLW2901 --isolated issue.py
issue.py:3:5: PLW2901 `for` loop variable `mutable` overwritten by assignment target
Found 1 error.
```

In this case, the augmented assignment is performed in-place and is equivalent to calling `extend` (which does not trigger PLW2901).

```
ruff --version
ruff 0.0.287
```

**Comment by charliermarsh** [comment]
Hmm, I think there are reasonable arguments both ways but I'm going to opt to leave this as-is for now. Pylint has the same behavior (`W2901: Redefining 'mutable' from loop (line 2) (redefined-loop-name)`), and the variable _is_ being reassigned here. This strikes me as the rule working as-intended.

**Comment by mihaic** [comment]
Thanks for your quick response. I find `redefined-loop-name` generally useful. Considering that Ruff behavior will stay the same, do you have any thoughts on dealing with similar code? I can think of using the method form of in-place augmented assignment, but verbosity is an issue, e.g., `set.symmetric_difference_update` vs. `^=`.

**Comment by charliermarsh** [comment]
My honest recommendation would be to use the in-place update methods. Even in the example above, I find the augmented assignment a bit less clear, since, e.g., _this_ doesn't work:

```python
mutables = ([0, 1, 2], [10, 11, 12])
for mutable in mutables:
    mutable = mutable + [100]
```

In other words, it's relying on the fact that Python uses in-place operations for data types that support it.

Alternatively, you could assign `mutable` to a temporary variable to avoid the lint error, or just use a `noqa` in these cases.
